modules/web-discovery-project/sources/web-discovery-project.es

```diff
@@ -1908,28 +1925,37 @@ const WebDiscoveryProject = {
           return true;
       } else {
         if (
+          !options.allowlisted &&
           url_parts.query_string &&
           url_parts.query_string.length > WebDiscoveryProject.qs_len
-        )
+        ) {
+          _log('DLU failed: length of query string is longer than qs_len');
           return true;
+        }
 
         if (url_parts.query_string) {
           var v = url_parts.query_string.split(/[&;]/);
           if (v.length > 4) {
             // that means that there is a least one &; hence 5 params
+            _log('DLU failed: there are more than 4 parameters');
             return true;
           }
           if (
+            !options.allowlisted &&
             WebDiscoveryProject.checkForLongNumber(
               url_parts.query_string,
               12
             ) != null
-          )
+          ) {
+            _log('DLU failed: long number in the query string: ' + url_parts.query_string);
```

Comment:
Nit. No need to concatenate if `log_` is not doing anything (log disabled). Also, even though JavaScript will not complain about it, `query_string` can be null and it sounds wrong to concatenate `null` to a string. 
```suggestion
            _log('DLU failed: long number in the query string: ' + url_parts.query_string);
```